src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -178,12 +204,13 @@ private void Reset(bool isDisposing = false)
 
             if (!isDisposing)
             {
+                _solutionLoadedTcs = new TaskCompletionSource<bool>();
```

Comment:
Actually `_solutionLoadedTcs?.TrySetCanceled();` statement is already over there at line 169, preceding this block in the same `Reset` method.